### PR TITLE
Add Yul smart contract language

### PR DESCRIPTION
**Pull request by davidhq**
Adding Yul --

"Yul is a programming language that is being used as an intermediate language in the compilation of Ethereum smart contracts written in in the Solidity language. However, Yul is designed to be more general than Solidity, and may find additional uses soon." Source: https://www.cs.utexas.edu/users/moore/acl2/manuals/current/manual/index-seo.php/YUL____YUL?path=3463/32691/2451/9091

Yul is becoming more and more important, here is next week's solidity summit agenda:
https://summit.soliditylang.org/agenda.html

There is quite a lot of .yul code on GitHub already.

## Checklist:
- [X] **I am adding a new language.**
  - [X] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Ayul+NOT+nothack
  - [X] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
      - https://docs.soliditylang.org/en/v0.8.13/yul.html#specification-of-yul-object (Specification of Yul Object)
      - https://docs.soliditylang.org/en/v0.8.13/yul.html#complete-erc20-example (Complete ERC20 Example)
    - Sample license(s):
    - © Copyright 2016-2021, Ethereum. (visible at the bottom of docs page)
    - GPL 3.0 - license of source repo
  - [X] I have included a syntax highlighting grammar: https://github.com/davidhq/SublimeEthereum
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -7223,6 +7223,15 @@ Yacc:
...
   color: "#4B6C4B"
   language_id: 409
+Yul:
+  type: programming
+  color: "#794932"
+  ace_mode: text
+  tm_scope: source.yul
+  extensions:
+  - ".yul"
+  - ".yulp"
```
Comment (lildude) [inline lib/linguist/languages.yml:7233]: There are only 52 `.yulp` files so nowhere near enough.

```suggestion
```

**Review by lildude — CHANGES_REQUESTED** [review]
Please remove the `.yulp` extension.

Please also update the OP to link to the direct source of the two samples and state their respective licenses.

**Comment by davidhq** [comment]
> Please remove the `.yulp` extension.
> 

done

> Please also update the OP to link to the direct source of the two samples and state their respective licenses.

Added direct links to samples, entire license is not present, all I could include is "© Copyright 2016-2021, Ethereum." visible at the bottom of the page.

If license for samples in longer form is needed I can write the team and ask if they can provide this? Please let me know if needed. Thank you

**Comment by lildude** [comment]
> If license for samples in longer form is needed I can write the team and ask if they can provide this? Please let me know if needed. Thank you

This is fine, thanks. I've found from the source that the docs are in a repo covered by the GPL 3.0 so I've updated your OP with that.

**Review by lildude — APPROVED** [review]

**Comment by davidhq** [comment]
> > If license for samples in longer form is needed I can write the team and ask if they can provide this? Please let me know if needed. Thank you
> 
> This is fine, thanks. I've found from the source that the docs are in a repo covered by the GPL 3.0 so I've updated your OP with that.

great, thank you!

**Pull request merged**